app/code/core/Mage/Catalog/Helper/Image.php

```diff
@@ -639,7 +639,13 @@ public function validateUploadFile($filePath) {
         }
 
         $_processor = new Varien_Image($filePath);
-        return $_processor->getMimeType() !== null;
+        $result = $_processor->getMimeType() !== null;
```

Comment:
None of this is needed. When the method returns the $_processor object will be destroyed automatically and trigger the __destruct() method. That method currently will trigger a call to close() you added and really should just have the close() code directly embedded.